api/types/asset.go

```diff
@@ -0,0 +1,71 @@
+// Copyright © 2023 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"fmt"
+	"strings"
+)
+
+func (a *VMInfoArchitecture) UnmarshalText(text []byte) error {
+	var arch VMInfoArchitecture
+
+	switch string(text) {
+	case "x86_64":
+		arch = Amd64
+	case "arm64":
+		arch = Arm64
+	default:
+		return fmt.Errorf("failed to unmarshal text into VMInfoArchitecture: %s", text)
+	}
+
+	*a = arch
+
+	return nil
+}
+
+func (a *VMInfoArchitecture) MarshalText() (string, error) {
+	switch *a {
+	case Amd64:
+		return "x86_64", nil
+	case Arm64:
+		return "arm64", nil
+	case Unknown:
+		return "", fmt.Errorf("unknown VMInfoArchitecture: %v", *a)
+	default:
+		return "", fmt.Errorf("failed to marshal VMInfoArchitecture into text: %v", *a)
+	}
+}
+
+type FromArchitectureMapping map[string]string
+
+func (m *FromArchitectureMapping) UnmarshalText(text []byte) error {
```

Comment:
the issue here is that mapstructure will continue recursively unmarshalling keys and values once the map has been loaded (which happened here in byte unmarshaller), leading to runtime panic issues because keys are proper type `VMInfoArchitecture` but they will anyway be casted to string